src/FindARoommate/FindARoommate.js

```diff
@@ -1,31 +1,71 @@
 import React, { useState, useEffect } from 'react';
-import getAllResidents from '../apiCalls';
-import ControlledOpenSelect from '../Form/Form';
-import SingleLineGridList from '../Thumbnails/Thumbnails'
+import {getAllResidents, getResidentsBySemester} from '../apiCalls';
+import Form from '../Form/Form';
+import Thumbnails from '../Thumbnails/Thumbnails'
+import ResidentCard from '../Profile/Profile';
+import { makeStyles } from '@material-ui/core/styles';
+import Box from '@material-ui/core/Box';
+import Container from '@material-ui/core/Container';
+
+const useStyles = makeStyles({
+  form: {
+    marginTop:-20
+  },
+  text: {
+    color: "#7c8181"
+  },
+  box: {
+    paddingTop: 50, 
+    paddingBottom: 80
+  }
+});
 
 export default function FindARoomate() {
+  const classes = useStyles();
+
   const [semester, setSemester] = useState('');
   const [residents, setResidents] = useState([]);
+  const [availableResidents, setAvailableResidents] = useState([]);
+  const [selectedResident, setSelectedResident] = useState('');
 
   useEffect(() => {
-    getResidents();
+    if (!availableResidents.length) {
+      getResidents();
+    } else {
+      getAvailableResidents(semester);
+    }
   }, [])
 
   const getResidents = async () => {
     await getAllResidents()
     .then(data => setResidents(data))
     .catch(err => console.log(err))
-    console.log(residents)
   }
 
   const selectSemester = (selectedSemester) => {
     setSemester(selectedSemester)
+    getAvailableResidents(semester)
+  }
+
+  const getAvailableResidents = async (semester) => {
+    await getResidentsBySemester(semester)
+    .then(data => setAvailableResidents(data))
+    .catch(err => console.log(err))
   }
   
+  const selectResident = (resident) => {
+    setSelectedResident(resident)
+    console.log(resident)
+  }
+
   return(
-    <div>
-      <ControlledOpenSelect selectSemester={selectSemester}/>
-      <SingleLineGridList availableResidents={residents} semesterAvailable={semester}/>
-    </div>
+    <Container>
+      <Box className={classes.box}>
+      <h1 className={classes.text}>Select a semester:</h1>
+      <Form className={classes.form} selectSemester={selectSemester}/>
+      </Box>
+      <Thumbnails availableResidents={availableResidents} allResidents={residents} selectResident={selectResident} semesterAvailable={semester}/>
+      {selectedResident && <ResidentCard resident={selectedResident} />}
```

Comment:
React and Material-ui has brought a lot of power to our project.

Reply:
Sure has! I added conditional rendering here for the ResidentCard so that we can style it as we develop, not for final production